agent/snmp_agent.c

```diff
@@ -3771,7 +3771,7 @@ handle_pdu(netsnmp_agent_session *asp)
     case SNMP_MSG_INTERNAL_SET_RESERVE1:
 #endif /* NETSNMP_NO_WRITE_SUPPORT */
         asp->vbcount = count_varbinds(asp->pdu->variables);
-        asp->requests = calloc(asp->vbcount, sizeof(netsnmp_request_info));
+        asp->requests = calloc(asp->vbcount+1, sizeof(netsnmp_request_info));
```

Comment:
If memory is allocated for one varbind more than needed, that will prevent Valgrind from detecting certain out-of-bounds errors. Can `asp->vbcount + 1` be changed into `asp->vbcount ? asp->vbcount : 1`?